# The demonstration of the tropical precipitation module on the different datasets

###  Loading the necessary packages

In [1]:
import sys

### Setting the path to the diagnostic repository 

In [3]:
sys.path.append('..')

In [ ]:
sys.path

In [9]:
#dask.__version__

'2023.3.1'

In [2]:
configdir = '../../../config/'
diagname  = 'tr_pr'
machine   = 'levante'

### Importing the aqua module 

In [3]:
import aqua

In [4]:
from aqua import Reader
from aqua.reader import catalogue

In [5]:
catalogue(configdir=configdir)

IFS	tco3999-ng5	2.8km experiment, coupled with FESOM
	- ICMGG_atm2d	
	- ICMU_atm2d	
	- ICMU_atm3d	
	- interpolated_global	
	- interpolated_np	
	- interpolated_sp	
	- interpolated_sp_ci	
	- lra-r100-day	
	- lra-r100-mon	
IFS	tco2559-ng5	4km experiment, coupled with FESOM
	- ICMGG_atm2d	
	- ICMU_atm2d	
	- ICMU_atm3d	
	- interpolated_global	
	- interpolated_np	
	- interpolated_sp	
	- lra-r100-day	
	- lra-r100-mon	
IFS	tco1279-orca025	9km baseline, coupled to NEMO, deep conv ON
	- ICMGG_atm2d	
	- ICMU_atm2d	
	- ICMU_atm3d	
	- lra-r100-day	
	- lra-r100-mon	
IFS	test-tco2559	4km experiment, coupled with FESOM
	- ICMGG_atm2d	2d output
	- ICMU_atm3d	3d output

FESOM	tco3999-ng5	2.5km experiment, coupuled with IFS
	- elem_grid	
	- node_grid	
	- np	nearest-neighbor interpolation to lat-lon grid
	- interpolated_global2d	
	- interpolated_global_TS	
	- interpolated_global_UV	
	- interpolated_np	
	- interpolated_sp	
	- original_2d	original 2d output
	- original_3d	original 3d output
FESOM	tco2559-ng

levante:
  args:
    path: ../../../config//levante/catalog.yaml
  description: ''
  driver: intake.catalog.local.YAMLFileCatalog
  metadata: {}


# DATA TESTED: 
    + ICON 
    + MSWEP 
    + IFS
    - FESOM (problems with datatype)
    - CERES (which variable)
    - ERA5 (got an unexpected keyword argument 'metadata')

# Successful TESTS

## ICON data

In [6]:
reader = Reader(model="ICON", exp="ngc2009",  configdir=configdir, source="atm_2d_ml_R02B09", regrid="r200")
ICON_2009_reg = reader.retrieve()

In [9]:
ICON_2009_reg['tprate']

<xarray.DataArray 'tprate' (time: 37009, cell: 20971520)>
dask.array<mul, shape=(37009, 20971520), dtype=float32, chunksize=(6, 1048576), chunktype=numpy.ndarray>
Coordinates:
  * time     (time) datetime64[ns] 2020-01-20 2020-01-20T00:30:00 ... 2022-03-01
Dimensions without coordinates: cell
Attributes: (12/16)
    CDI_grid_type:                unstructured
    cell_methods:                 t: mean
    long_name:                    Mean total precipitation rate
    number_of_grid_in_reference:  1
    param:                        52.1.0
    standard_name:                pr
    ...                           ...
    shortName:                    tprate
    target_units:                 m s**-1
    factor:                       0.001
    offset:                       0
    src_units:                    kg m-2 s-1
    units_fixed:                  1

## MSWEP

In [8]:
reader = Reader(model="MSWEP", exp="past", source="monthly",configdir=configdir)
#data = reader.retrieve(fix=False)
#data = reader.retrieve(apply_unit_fix=False)
MSWEP = reader.retrieve()

In [10]:
MSWEP['tprate'] 

<xarray.DataArray 'tprate' (time: 502, lat: 1800, lon: 3600)>
dask.array<mul, shape=(502, 1800, 3600), dtype=float32, chunksize=(1, 1800, 3600), chunktype=numpy.ndarray>
Coordinates:
  * lon      (lon) float32 -179.9 -179.8 -179.8 -179.6 ... 179.8 179.9 179.9
  * lat      (lat) float32 89.95 89.85 89.75 89.65 ... -89.75 -89.85 -89.95
  * time     (time) datetime64[ns] 1979-02-01 1979-03-01 ... 2020-11-01
Attributes:
    units:         m s**-1
    history:       2023-03-31 03:01:26 variable renamed by AQUA fixer;\n2023-...
    paramId:       172228
    long_name:     Mean total precipitation rate
    cfVarName:     tprate
    shortName:     tprate
    target_units:  m s**-1
    factor:        3.802570537683474e-10
    offset:        0
    src_units:     mm month-1
    units_fixed:   1

## IFS

In [11]:
reader = Reader(model="IFS", exp="tco3999-ng5", source="ICMGG_atm2d",configdir=configdir)
#data = reader.retrieve(fix=False)
#data = reader.retrieve(apply_unit_fix=False)
IFS = reader.retrieve()

In [12]:
IFS['tprate']

<xarray.DataArray 'tprate' (time: 5401, value: 64144000)>
dask.array<mul, shape=(5401, 64144000), dtype=float64, chunksize=(1, 64144000), chunktype=numpy.ndarray>
Coordinates:
    lat      (value) float64 dask.array<chunksize=(64144000,), meta=np.ndarray>
    lon      (value) float64 dask.array<chunksize=(64144000,), meta=np.ndarray>
  * time     (time) datetime64[ns] 2020-01-20 2020-01-20T01:00:00 ... 2020-09-01
Dimensions without coordinates: value
Attributes: (12/23)
    paramId:                    172228
    dataType:                   fc
    numberOfPoints:             64144000
    typeOfLevel:                surface
    stepUnits:                  1
    stepType:                   instant
    ...                         ...
    target_units:               m s**-1
    factor:                     0.0002777777777777778
    offset:                     0
    decumulated:                1
    src_units:                  m
    units_fixed:                1

In [19]:
# ERA5
#import intake_esm
#intake_esm.__version__

'2021.8.17'

# NOT successful TESTS

# Doesn`t work
Error: ValueError: conflicting sizes for dimension 'cell': length 20971520 on  and length 327680 on {'cell': 'cell'}

In [ ]:
reader = Reader(model="ICON", exp="ngc2009",configdir=configdir,  source="atm_2d_ml_R02B06")
#retrieve(fix=False) OR retrieve(apply_unit_fix=False)
ICON_2009 = reader.retrieve()

In [ ]:
# Add units of x axe
# name of the model ...
# and etc

# NOT successful TESTS

# ERA5

Error: TypeError: 'NoneType' object is not iterable

In [ ]:
reader = Reader(model="ERA5", exp="era5", source="monthly_inst",configdir=configdir)
ERA5_data = reader.retrieve()

In [ ]:
reader = Reader(model='ERA5',exp='era5',source='monthly_inst',configdir=configdir,regrid='r100')
ERA5_reg = reader.retrieve()

# FESOM

Erro: ValueError: conflicting sizes for dimension 'lat': length 2700 on  and length 2000 on {'lon': 'lon', 'lat': 'lat'}

In [ ]:

reader = Reader(model="FESOM", exp="tco3999-ng5", source="np", configdir=configdir)
#data = reader.retrieve(fix=False)
#data = reader.retrieve(apply_unit_fix=False)
FESOM = reader.retrieve()

# CERES

Error: TypeError: 'NoneType' object is not iterable

In [ ]:
#reader = Reader(model="CERES", exp="syn-toa41", source="daily")
#reader = Reader(model="CERES", exp="ebaf-sfc41", source="monthly")
reader = Reader(model="CERES", exp="ebaf-toa42", source="monthly",configdir=configdir)
#reader = Reader(model="CERES", exp="ebaf-toa41 ", source="monthly")
CERES = reader.retrieve()